### PR TITLE
Enable searching on pressing enter in the search box

### LINKED ISSUE #440: Run search after keyboard enter press

### LINKED ISSUE #442: Bugs with search and changing workspace
1. Keep 2 workspaces ready:
Personal - Have some plios
AF - Have no plios

In personal workspace, search something which gives no results -> we see the no plios found table

Now switch the workspace -> we see AF's workspace with no plios and the starting screen

Switch back to personal, it still shows the same no plios and the starting screen.

2. - Keep 2 workspaces ready
Personal - Have some plios
AF - Have no plios

In personal workspace, search something which gives SOME results -> we see the results filtered and shown to us

Now switch the workspace -> we see AF's workspace with no plios and the starting screen

Switch back to personal, it still shows the filtered results but in the search bar there is no string
- either we display that string there somehow
- or we dont' show them the filtered table and show them the default table

3. If you have 2 workspaces, both of which have some plios: if you search something in one workspace, then switch workspaces, the search string still remains there.

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #440 
Fixes #442 

## Test Plan

- [ ] ~Test Responsiveness~
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [x] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari
- [ ] ~Local Language Support~
- [x] Wrote tests
- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on an actual physical phone
- [ ] Tested on production
- [x] Comments have been added appropriately
- [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
- [x] Added relevant details like Labels/Projects/Milestones etc.

### CODE REVIEW

File: src/components/Collections/Table/Table.vue [hunk 1]
```diff
@@ -292,6 +293,20 @@ export default {
...
+       */
+      // check if the key pressed is the enter key
+      if (event.key === "Enter" || event.keyCode === 13) {
+        /**
+         * event.key is the modern way of detecting keys
+         * event.keyCode is deprecated (left here for for legacy browsers support)
+         */
+        if (this.searchString != "") this.search();
+      }
+    },
```
Comment (deepansh96) [inline src/components/Collections/Table/Table.vue:309]: Are we sure that in legacy browsers, `event.key` will not throw an error like "property key of undefined not found"?

File: src/components/Collections/Table/Table.vue [hunk 2]
```diff
@@ -292,6 +293,20 @@ export default {
...
+       * detect if enter has been pressed after entering
+       * a text to search
+       */
+      // check if the key pressed is the enter key
+      if (event.key === "Enter" || event.keyCode === 13) {
+        /**
+         * event.key is the modern way of detecting keys
+         * event.keyCode is deprecated (left here for for legacy browsers support)
+         */
+        if (this.searchString != "") this.search();
```
Comment (deepansh96) [inline src/components/Collections/Table/Table.vue:307]: I think we should also check for leading or trailing spaces here and if they're present, trim those before proceeding with this check

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/components/Collections/Table/Table.vue (hunk 1 above)
Comment (dalmia) [reply src/components/Collections/Table/Table.vue:309]: haan woh `event.key` sab mein hi hota hain from what I know!

File: src/components/Collections/Table/Table.vue (hunk 2 above)
Comment (dalmia) [reply src/components/Collections/Table/Table.vue:307]: makes sense

**Pull request merged**